lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Action.js

```diff
@@ -242,55 +242,75 @@ qx.Class.define("callbackery.ui.plugin.Action", {
                 }
             },this);
         },
-        _addUploadButton: function(cfg,btCfg,getFormData){
-            var form = new uploadwidget.UploadForm('uploadFrm','upload');
-            form.setParameter('name',cfg.name);
-            form.setParameter('key',btCfg.key);
-            form.setLayout(new qx.ui.layout.HBox());
-            this.add(form);
-            var file;
+        _makeUploadButton: function(cfg,btCfg,getFormData){
+            
+            
+            var button;
             if (btCfg.btnClass == 'toolbar') {
-                file = new uploadwidget.UploadToolbarButton('file', this.xtr(btCfg.label));
+                button = new callbackery.ui.form.UploadToolbarButton(
+                    this.xtr(btCfg.label));
             }
             else {
-                file = new uploadwidget.UploadButton('file', this.xtr(btCfg.label));
+                button = new callbackery.ui.form.UploadButton(
+                    this.xtr(btCfg.label));
             }
             if (btCfg.key){
-                this._buttonMap[btCfg.key]=file;
+                this._buttonMap[btCfg.key]=button;
             }
             if (btCfg.buttonSet) {
-                file.set(btCfg.buttonSet);
+                button.set(btCfg.buttonSet);
                 if (btCfg.key){
                     this._buttonSetMap[btCfg.key]=btCfg.buttonSet;
                 }
             }
-            form.add(file);
-            file.addListener('execute',function(e){
+            var serverCall = callbackery.data.Server.getInstance();
+            button.addListener('changeSelection',function(e){
```

Comment:
should be `button.addListener('changeFileName',function(e){`

Reply:
I changed the name